.github/workflows/build-push-image.yml

```diff
@@ -0,0 +1,38 @@
+name: Build and Push Docker Image
+
+on:
+  pull_request:
+    types: [closed]
+    branches:
+      - main
```

Comment:
I think we should remove that: otherwise when a PR is closed but not merged, the workflow runs (and let's imagine it's because there are issues with it, we'll be in trouble)